ledger-core/virtual/integration/delegationtoken_test.go

```diff
@@ -225,3 +225,89 @@ func TestDelegationToken_CheckFailIfWrongApprover(t *testing.T) {
 	}
 	mc.Finish()
 }
+
+func TestDelegationToken_CheckFailIfSenderEqApprover(t *testing.T) {
+	t.Log("C5193")
+	cases := []struct {
+		name string
+		msg  interface{}
+	}{
+		{
+			name: "VCallRequest",
+			msg:  &payload.VCallRequest{},
+		},
+		{
+			name: "VCallResult",
+			msg:  &payload.VCallResult{},
+		},
+		{
+			name: "VStateReport",
+			msg:  &payload.VStateReport{},
+		},
+		{
+			name: "VStateRequest",
+			msg:  &payload.VStateRequest{},
+		},
+		{
+			name: "VDelegatedCallRequest",
+			msg:  &payload.VDelegatedCallRequest{},
+		},
+		{
+			name: "VDelegatedRequestFinished",
+			msg:  &payload.VDelegatedRequestFinished{},
+		},
+	}
+
+	mc := minimock.NewController(t)
+
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			server, ctx := utils.NewUninitializedServer(nil, t)
+
+			jetCoordinatorMock := jet.NewAffinityHelperMock(t)
+			auth := authentication.NewService(ctx, jetCoordinatorMock)
+			server.ReplaceAuthenticationService(auth)
+
+			var errorFound bool
+			logHandler := func(arg interface{}) {
+				if err, ok := arg.(error); ok {
+					errMsg := err.Error()
+					if severity, sok := throw.GetSeverity(err); sok && severity == throw.FraudSeverity &&
+						strings.Contains(errMsg, "sender cannot be approver of the token") &&
+						strings.Contains(errMsg, "illegitimate msg") {
+						errorFound = true
+					}
+				}
+			}
+			logger := utils.InterceptLog(inslogger.FromContext(ctx), logHandler)
+			server.OverrideConveyorFactoryLogContext(inslogger.SetLogger(ctx, logger))
+
+			server.Init(ctx)
+			// increment pulse for VStateReport and VDelegatedCallRequest
+			server.IncrementPulse(ctx)
+
+			jetCoordinatorMock.
+				MeMock.Return(server.GlobalCaller()).
+				QueryRoleMock.Set(
+				func(_ context.Context, _ node.DynamicRole, obj reference.Local, pulse pulse.Number) ([]reference.Global, error) {
```

Comment:
May be written shorter
QueryRoleMock.Return([]reference.Global{server.GlobalCaller()}, nil)

Reply:
Done